src/common/main/kotlin/me/owdding/skyocean/features/inventory/museum/MuseumDonationHelper.kt

```diff
@@ -0,0 +1,338 @@
+import me.owdding.skyocean.utils.Utils.skyoceanReplace
+import me.owdding.skyocean.utils.Utils.unaryPlus
+import me.owdding.skyocean.utils.Utils.wrap
+import net.minecraft.client.gui.components.AbstractWidget
+import net.minecraft.client.gui.screens.inventory.tooltip.ClientTextTooltip
+import net.minecraft.client.gui.screens.inventory.tooltip.ClientTooltipComponent
+import net.minecraft.network.chat.Component
+import net.minecraft.world.item.Item
+import net.minecraft.world.item.ItemStack
+import net.minecraft.world.item.Items
+import tech.thatgravyboat.skyblockapi.api.datatype.DataTypes
+import tech.thatgravyboat.skyblockapi.api.events.base.Subscription
+import tech.thatgravyboat.skyblockapi.api.events.base.predicates.MustBeContainer
+import tech.thatgravyboat.skyblockapi.api.events.base.predicates.OnlyOnSkyBlock
+import tech.thatgravyboat.skyblockapi.api.events.screen.ContainerCloseEvent
+import tech.thatgravyboat.skyblockapi.api.events.screen.InventoryChangeEvent
+import tech.thatgravyboat.skyblockapi.helpers.McFont
+import tech.thatgravyboat.skyblockapi.helpers.McScreen
+import tech.thatgravyboat.skyblockapi.utils.extentions.cleanName
+import tech.thatgravyboat.skyblockapi.utils.extentions.get
+import tech.thatgravyboat.skyblockapi.utils.text.TextBuilder.append
+import tech.thatgravyboat.skyblockapi.utils.text.TextColor
+import tech.thatgravyboat.skyblockapi.utils.text.TextProperties.stripped
+import tech.thatgravyboat.skyblockapi.utils.text.TextStyle.color
+
+@Module
+@LoreModifier
+object MuseumDonationHelper : MeowddingLogger by SkyOcean.featureLogger(), RecipeView, AbstractLoreModifier() {
+
+    private val modifierCache: MutableMap<String, Pair<ComponentModifier?, TooltipComponentModifier?>> = mutableMapOf()
+
+    private context(item: ItemStack) fun registerModifier(component: ComponentModifier?, tooltip: TooltipComponentModifier?) {
+        if (component == null && tooltip == null) return
+        modifierCache[item.cleanName] = component to tooltip
+    }
+
+    private context(item: ItemStack) fun registerModifier(component: ComponentModifier) {
+        modifierCache[item.cleanName] = component to null
+    }
+
+    private context(item: ItemStack) fun buildModifiers(init: ModifierBuilder.() -> Unit) {
+        val builder = object : ModifierBuilder {
+            override var component: ComponentModifier? = null
+            override var tooltip: TooltipComponentModifier? = null
+
+            override fun registerModifier(component: ComponentModifier) {
+                this.component = component
+            }
+
+            override fun registerComponentModifier(modifier: TooltipComponentModifier) {
+                this.tooltip = modifier
+            }
+        }
+        builder.init()
+        if (builder.component == null && builder.tooltip == null) return
+        modifierCache[item.cleanName] = builder.component to builder.tooltip
+    }
+
+    val museumRegex = Regex(".*?[Mm]useum.*?")
```

Comment:
```suggestion
    val museumRegex = Regex("[Mm]useum")
```

can be thsi and use find instead of matches, can also exclude the Mm and turn on case insensitivity